test/src/org/labkey/test/tests/targetedms/TargetedMSOutlierHeatmapTest.java

```diff
@@ -0,0 +1,139 @@
+        init.doSetup();
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest)
+    {
+        _containerHelper.deleteProject(getProjectName(), afterTest);
+    }
+
+    private void doSetup()
+    {
+        setupFolder(FolderType.QC);
+        importData(SProCoP_FILE);
+
+        PortalHelper _portalHelper = new PortalHelper(getDriver());
+        _portalHelper.enterAdminMode();
+        _portalHelper.deleteTab("Raw Data");
+        _portalHelper.deleteTab("Pareto Plot");
+        _portalHelper.deleteTab("Annotations");
+        _portalHelper.addTab(OUTLIER_HEATMAP_TAB);
+
+        _portalHelper.addWebPart("Outlier Heatmap Summary View");
+        _portalHelper.exitAdminMode();
+    }
+
+    @Test
+    public void testHeatMapColorAndValues() throws InterruptedException
+    {
+        log("Updating the Metric type values");
+        goToProjectHome();
+        QCPlotsWebPart qcPlotsWebPart = new PanoramaDashboard(this).getQcPlotsWebPart();
+        qcPlotsWebPart.clickMenuItem("Configure QC Metrics");
+        ConfigureMetricsUIPage configureQCMetrics = new ConfigureMetricsUIPage(this);
+        configureQCMetrics.setFixedDeviationFromMean(QCPlotsWebPart.MetricType.PRECURSOR_AREA, "-5", "5")
+                .clickSave();
+
+        clickPortalTab(OUTLIER_HEATMAP_TAB);
+        OutlierHeatmapSummaryWebPart outlierHeatmap = new OutlierHeatmapSummaryWebPart(getDriver());
+        Assert.assertEquals("Incorrect outlier count", "2",
+                outlierHeatmap.getCellElement(1, QCPlotsWebPart.MetricType.FWHM).getText());
+        Assert.assertEquals("Incorrect total replicate count", "47", outlierHeatmap.getTotalReplicateCount().trim());
+
+        log("Verify data range: " + OutlierHeatmapSummaryWebPart.HeatmapDateRange.Last_7_Days);
+        outlierHeatmap.setDateRange(OutlierHeatmapSummaryWebPart.HeatmapDateRange.Last_7_Days, true);
+        Assert.assertEquals("Incorrect outlier count for " + QCPlotsWebPart.MetricType.FWHM, "1",
+                outlierHeatmap.getCellElement(1, QCPlotsWebPart.MetricType.FWHM).getText());
+
+        log("Verify heatmap colors");
+        Assert.assertTrue("Incorrect heatmap color for highest(Red)",
+                outlierHeatmap.getCellElement(1, QCPlotsWebPart.MetricType.PRECURSOR_AREA).getAttribute("style").contains("background-color: rgb(255, 0, 0)"));
+        Assert.assertTrue("Incorrect heatmap color for lowest(White)",
+                outlierHeatmap.getCellElement(1, QCPlotsWebPart.MetricType.ISOTOPE_DOTP).getAttribute("style").contains("background-color: rgb(255, 255, 255)"));
+
+        log("Verify Custom date range");
+        outlierHeatmap.setDateRange(OutlierHeatmapSummaryWebPart.HeatmapDateRange.Custom_Range, false);
+        outlierHeatmap.doAndWaitForTableUpdate(() -> {
+            outlierHeatmap.setStartDate("2013-08-01");
+            outlierHeatmap.setEndDate("2013-08-15");
+        });
```

Comment:
I would have a separate method for custom date range. If possible, tests shouldn't need to know when an update is expected.
```suggestion
        outlierHeatmap.setCustomDateRange("2013-08-01", "2013-08-15");
```